### ISSUE TITLE
Not able to stream RTSP on IOS. EXC_BAD_ACCESS when run

### ISSUE THREAD

**Issue by CVHOANG-coder** [razorRun/react-native-vlc-media-player#208]
I am trying to live stream using RTSP, the url is wokring fine in the actual media player but it's not working and show error:  EXC_BAD_ACCESS when trying to stream it using VLCPlayer. Both simulator and real device

![image](https://github.com/razorRun/react-native-vlc-media-player/assets/55902338/1c7ea5bd-7214-45cc-8d3b-3eb6a8350035)

**Comment by GustavoBonfimS** [comment]
any solution ?

**Comment by GustavoBonfimS** [comment]
I faced this issue building in an Mac M1. The app crashes on simulator but works fine in real device.
The temporary solution for me was building with different arch using:
`yarn ios --simulator "iPhone 15" --destination arch=x86_64`

**Comment by CVHOANG-coder** [comment]
replace special symbol in url path. Example '@' to '%40'. valid url : encodeURI(), encodeURIComponent()

**Comment by hakanakin** [comment]
> I faced this issue building in an Mac M1. The app crashes on simulator but works fine in real device. The temporary solution for me was building with different arch using: `yarn ios --simulator "iPhone 15" --destination arch=x86_64`

npx react-native run-ios --simulator="iPhone 15 Pro" --destination arch=x86_64  that's work for me to thx

**Comment by roshansinghg** [comment]
> valid url : encodeURI(),

where to do this??

**Comment by tunm1228** [comment]
> encodeURIComponent

If your account or password contains special characters, use encodeURIComponent

**Comment by roshansinghg** [comment]
> replace special symbol in url path. Example '@' to '%40'. valid url : encodeURI(), encodeURIComponent(). I fix it

I am just getting rtmp live streaming link from backend like 
uri: "rtmp://a704d90f9059140b6aa1bc1e9a918f45-1788519902.ap-south-1.elb.amazonaws.com:1935/live/4db23812-6918-49ef-80ee-d02e64d6a6fb",
working fine for iOS 17.2 and lower but crashing above that.

**Comment by tunm1228** [comment]
> > replace special symbol in url path. Example '@' to '%40'. valid url : encodeURI(), encodeURIComponent(). I fix it
> 
> I am just getting rtmp live streaming link from backend like uri: "rtmp://a704d90f9059140b6aa1bc1e9a918f45-1788519902.ap-south-1.elb.amazonaws.com:1935/live/4db23812-6918-49ef-80ee-d02e64d6a6fb", working fine for iOS 17.2 and lower but crashing above that.

Your path has no special characters, so this is not the case I encountered.Please learn more.

**Comment by diegods-ferreira** [comment]
I'm facing the same problem. Enconding the URL as @tunm1228 sugested did not work for me. My RTSP camera stream URL is "rtsp://admin:econdos123@192.168.1.123:554/Streaming/channels/101".

**Comment by fredrikburmester** [comment]
I'm trying to stream HLS and I get the same crash.

**Comment by hakanakin** [comment]
Hey, if you’re using <TouchableWithoutFeedback></TouchableWithoutFeedback> and your app crashes, you should replace it with a different option.

**Comment by fredrikburmester** [comment]
I can say that I'm **not** using TouchableWithoutFeedback and it is still crashing for me.

**Comment by diegods-ferreira** [comment]
I was able to solve my problem with RTSP streaming. It turns out I had to request for Apple's **Multicast Networking Entitlement** permission so my app could stream. Apparently this had been going since iOS 17 and I just now realised. But once I got permission for multicast networking my app stopped crashing.

For anyone who wants to know how to request this permission, here it is: https://forums.developer.apple.com/forums/thread/663271?answerId=639455022#639455022
